src/components/Table/Table.tsx

```diff
@@ -56,27 +99,44 @@ export function Table<T extends { id: number }>(props: TableProps<T>) {
     }));
   };
 
-  const filters = props.columns.map((column, index) => (
-    <td key={index}>
-      <TextInput
-        placeholder={`Filter by ${column.header}`}
-        onChange={(event) => {
-          handleFilterChange(event, column.columnName);
-        }}
-      />
-    </td>
-  ));
-
-  const adjustedData = useMemo(() => {
-    const filteredData = filterData(
-      [...props.data],
-      filtersState,
-      props.columns
-    );
-    const sortedData = sortData(filteredData, sortState);
+  const filters = props.columns.map(
+    (column, index) =>
+      column.isFilterable && (
+        <td key={index}>
+          <TextInput
+            placeholder={`Filter by ${column.header}`}
+            onChange={(event) => {
+              handleFilterChange(event, column.columnName);
+            }}
+          />
+        </td>
+      )
+  );
 
-    return sortedData;
-  }, [filtersState, sortState, props.data]);
+  const filteredData = useMemo(
+    () => filterData([...props.data], filtersState, props.columns),
+    [filtersState, props.data]
+  );
+
+  const sortedData = useMemo(
+    () => sortData(filteredData, sortState),
+    [filteredData, sortState]
+  );
+
+  const paginatedData = useMemo(() => {
+    const numberOfRows = Math.ceil(adjustedTableHeight / props.rowHeight);
```

Comment:
missing `props.rowHeight` in deps array, also `filtersState` isn't used thus redundant. Check other deps arrays also